shared/src/types/vote_extensions/validator_set_update.rs

```diff
@@ -171,10 +189,11 @@ impl VotingPowersMapExt for VotingPowersMap {
             .map(|&(_, &voting_power)| u64::from(voting_power))
             .sum();
 
-        // split the vec into two
-        sorted
-            .into_iter()
-            .map(|(&EthAddress(addr), &voting_power)| {
+        // split the vec into three portions
+        let init = (Vec::new(), Vec::new(), Vec::new());
+        sorted.into_iter().fold(
+            init,
+            |accum, (ref addr_book, &voting_power)| {
```

Comment:
Is `ref` necessary here? If we use `ref`, then the type of `addr_book` is `&&EthAddress` which seems odd. Similar sort of thing for the `refs` on L182, I think they could be removed